src/clients/forward.h

```diff
@@ -9,15 +9,12 @@
 #ifndef SQUID_CLIENTS_FORWARD_H
 #define SQUID_CLIENTS_FORWARD_H
 
+#include "base/forward.h"
 #include "sbuf/forward.h"
 
 class FwdState;
 class HttpRequest;
 
-class AsyncJob;
-template <class Cbc> class CbcPointer;
-typedef CbcPointer<AsyncJob> AsyncJobPointer;
```

Comment:
> PR description: Remove some duplicate pre-defines

I adjusted the PR title to use more accurate C++ terms for "pre-defines": "typedef and forward declarations". I do not insist on the PR title change. Please feel free to revert or adjust as needed.